plugins/output/adapt/importsource.js

```diff
@@ -1,109 +1,123 @@
-          max: bytes.format(form.maxFileSize),
-          size: bytes.format(form.bytesExpected)
-        })));
-      }
-      return done(error);
-    }
-    var formTags = (fields.tags && fields.tags.length) ? fields.tags.split(',') : [];
-    var formAssetDirs = (fields.formAssetFolders && fields.formAssetFolders.length) ? fields.formAssetFolders.split(',') : [];
-    var cleanFormAssetDirs = formAssetDirs.map(function(item) {
-      return item.trim();
+  /**
+  * Main process
+  * All functions delegated below for readability
+  */
+  async.series([
+    async.apply(prepareImport, files),
+    findLanguages,
+    validateCoursePackage,
+    installPlugins,
+    addAssets,
+    cacheMetadata,
+    importContent
+  ], function(importError, result) {
+    // cleanup should run regardless of import fail or success
+    helpers.cleanUpImport(cleanupDirs, function(cleanUpError) {
+      const error = importError || cleanUpError;
+      done(error);
     });
+  });
 
-
-    /**
-    * Main process
-    * All functions delgated below for readability
-    */
-    async.series([
-      function asyncUnzip(cb) {
-        helpers.unzip(files.file.path, COURSE_ROOT_FOLDER, function(error) {
-          if(error) return cb(error)
-          cleanupDirs.push(files.file.path, COURSE_ROOT_FOLDER);
-          cb();
+  function prepareImport(files, cb) {
+    async.parallel([
+      function(cb2) {
+        database.getDatabase(function(error, db) {
+          if(error) return cb2(error);
+          dbInstance = db; // cache this for reuse
+          cb2();
         });
       },
-      async.apply(findLanguages),
-      async.apply(validateCoursePackage, cleanFormAssetDirs),
-      async.apply(installPlugins),
-      async.apply(addAssets, formTags),
-      async.apply(cacheMetadata),
-      async.apply(importContent, formTags)
-    ], function(importError, result) {
-      // cleanup should run regardless of import fail or success
-      helpers.cleanUpImport(cleanupDirs, function(cleanUpError) {
-        const error = importError || cleanUpError;
-        done(error);
-      });
-    });
-  });
-
+      function(cb2) {
+        const form = new IncomingForm();
+        form.maxFileSize = configuration.getConfig('maxFileUploadSize');
+        // parse the form
+        form.parse(req, function (error, fields, files) {
+          if(error) {
+            if (form.bytesExpected > form.maxFileSize) {
+              return done(new Error(app.polyglot.t('app.uploadsizeerror', {
+                max: bytes.format(form.maxFileSize),
+                size: bytes.format(form.bytesExpected)
+              })));
+            }
+            return done(error);
+          }
+          var formAssetDirs = (fields.formAssetFolders && fields.formAssetFolders.length) ? fields.formAssetFolders.split(',') : [];
+          formTags = (fields.tags && fields.tags.length) ? fields.tags.split(',') : [];
+          cleanFormAssetDirs = formAssetDirs.map(item => item.trim());
```

Comment:
Is `cleanFormAssetDirs` used?

Reply:
Yeah, it's used in [this function](https://github.com/adaptlearning/adapt_authoring/blob/354ebc05e3c7637dcd38eeaac5f6be39cede60db/plugins/output/adapt/importsource.js#L169-L190)